### PR TITLE
cmd/regula: remove dependency on confita/backend/flags

### PR DESCRIPTION
**Pull request by rogpeppe**
The confita flags package is flawed and will probably
be removed in the future, so change to use the flag package
directly instead.

See https://github.com/heetch/confita/issues/63.

### CODE REVIEW

File: cmd/regula/cli/cli.go [hunk 1]
```diff
@@ -32,22 +34,50 @@ type Config struct {
...
 	var cfg Config
-	cfg.LogLevel = zerolog.DebugLevel.String()
+	flag := stdflag.NewFlagSet("", stdflag.ContinueOnError)
+	flag.StringVar(&cfg.Etcd.Namespace, "etcd-namespace", "", "etc namespace to use")
+	flag.StringVar(&cfg.LogLevel, "log-level", zerolog.DebugLevel.String(), "debug level")
 	cfg.Etcd.Endpoints = []string{"127.0.0.1:2379"}
-	cfg.Server.Address = "0.0.0.0:5331"
-	cfg.Server.Timeout = 5 * time.Second
-	cfg.Server.WatchTimeout = 30 * time.Second
+	flag.Var(commaSeparatedFlag{&cfg.Etcd.Endpoints}, "etcd-endpoints", "comma separated etc endpoints")
```
Comment (asdine) [inline cmd/regula/cli/cli.go]:
```suggestion
	flag.Var(commaSeparatedFlag{&cfg.Etcd.Endpoints}, "etcd-endpoints", "comma separated etcd endpoints")
```

**Comment by asdine** [comment]
@rogpeppe Could you change the base branch to `release-v0.7.0` please?

File: cmd/regula/cli/cli.go [hunk 2]
```diff
@@ -32,22 +34,50 @@ type Config struct {
...
+	flag.DurationVar(&cfg.Server.Timeout, "server-timeout", 5*time.Second, "server timeout (TODO)")
+	flag.DurationVar(&cfg.Server.WatchTimeout, "server-watch-timeout", 30*time.Second, "server watch timeout (TODO)")
 
-	err := confita.NewLoader(env.NewBackend(), flags.NewBackend()).Load(context.Background(), &cfg)
+	err := confita.NewLoader(env.NewBackend()).Load(context.Background(), &cfg)
 	if err != nil {
 		return nil, err
 	}
-
+	if err := flag.Parse(os.Args[1:]); err != nil {
```
Comment (YassineE) [inline cmd/regula/cli/cli.go]:
```suggestion
	if err := flag.Parse(args[1:]); err != nil {
```

Let's use the args received in parameters to ease testing.

File: cmd/regula/cli/cli.go [hunk 3]
```diff
@@ -32,22 +34,50 @@ type Config struct {
...
 
 // LoadConfig loads the configuration from the environment or command line flags.
-func LoadConfig() (*Config, error) {
+// The args hold the command line arguments, as found in os.Argv.
+// It returns flag.ErrHelp if the -help flag is specified on the command line.
+func LoadConfig(args []string) (*Config, error) {
 	var cfg Config
-	cfg.LogLevel = zerolog.DebugLevel.String()
+	flag := stdflag.NewFlagSet("", stdflag.ContinueOnError)
+	flag.StringVar(&cfg.Etcd.Namespace, "etcd-namespace", "", "etc namespace to use")
```
Comment (YassineE) [inline cmd/regula/cli/cli.go]:
```suggestion
	flag.StringVar(&cfg.Etcd.Namespace, "etcd-namespace", "", "etcd namespace to use")
```

**Comment by asdine** [comment]
@rogpeppe :up: https://github.com/heetch/regula/pull/113#issuecomment-493378766

**Comment by rogpeppe** [comment]
oops, missed your comment, sorry!

**Comment by YassineE** [comment]
@rogpeppe can you cherry-pick the commit to `release-v0.7.0` branch please?

**Comment by rogpeppe** [comment]
@YassineE done in #117.

**Pull request merged**